makefunc.go

```diff
@@ -0,0 +1,119 @@
+package makefunc
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var (
+	ErrorType = reflect.TypeOf((*error)(nil)).Elem()
+)
+
+// CheckTypeCompatibility checks whether `got` is a valid type for `want`, and
+// returns a descriptive error otherwise.
+//
+// Examples of positive compatibility:
+//   want=int, got=int
+//   want=interface{}, got=string
+//   want=interface{}, got=nil
+//   want=error, got=untyped nil
+//   want=error, got=error-typed nil
+//   want=io.ReadWriter, got=io.Writer
+//   want=[]string, got=[]string
+//   want=[]interface{}, got=[]string
```

Comment:
Heh. In usual life, this is an invalid assignment :)

Reply:
duh, right, I'll fix that